src/OpenTelemetry/Metrics/MeterProviderBuilderExtensions.cs

```diff
@@ -223,11 +219,45 @@ public static MeterProvider Build(this MeterProviderBuilder meterProviderBuilder
             return null;
         }
 
-        private static bool IsSortedAndDistinct(double[] values)
+        internal static bool CleanHistogramBounds(MetricStreamConfiguration metricStreamConfig, Action onInvalidFunction)
+        {
+            if (metricStreamConfig is ExplicitBucketHistogramConfiguration histogramConfiguration)
+            {
+                if (histogramConfiguration.Boundaries != null)
+                {
+                    if (histogramConfiguration.Boundaries.Length > 0)
+                    {
+                        if (!HasValidHistogramBoundaries(histogramConfiguration.Boundaries))
+                        {
+                            onInvalidFunction();
+                            return false;
+                        }
+
+                        // Remove any infinity values from the histogram boundaries
+                        histogramConfiguration.Boundaries = histogramConfiguration.Boundaries.Where(x => !double.IsInfinity(x)).ToArray();
+
+                        // If empty bounds, use (-inf, +inf) as the single bucket
+                        if (!Enumerable.Any(histogramConfiguration.Boundaries))
+                        {
+                            histogramConfiguration.Boundaries = new double[] { double.NegativeInfinity, double.PositiveInfinity };
+                        }
+                    }
+                }
+                else
+                {
+                    // Use default value when bounds are null
+                    histogramConfiguration.Boundaries = Metric.DefaultHistogramBounds;
```

Comment:
Do we need this? This should be taken care of in the `Metric` ctor: https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Metrics/Metric.cs#L113

Reply:
Tests seem to work without it :)